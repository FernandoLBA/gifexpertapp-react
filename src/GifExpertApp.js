import React, { Fragment, useState } from 'react';
import { AddCategory } from './components/AddCategory';
import Footer from './components/Footer';
import { GifGrid } from './components/GifGrid';
import './index.css';

// Creamos el functional component
export const GifExpertApp = () => {

     // lista de categorías:

     // Usamos un Hook
     // const [categories, setCategories] = useState( [ 'One Punch', 'Samurai X', 'Dragon ball Z' ] );

     const [categories, setCategories] = useState( [] );

     // TAREA
     // Al oprimir el botón agregar, debe agregar un nuevo 
     // elemento al arreglo.
     // const handleAdd = () => {
          
     //      const valor = 'Caballeros del Zodíaco';

     //      // En lugar de usar push, agregamos el nuevo valor de 
     //      // la siguiente manera, carga la copia del arreglo 
     //      // cateogies mediante el spread operator y luego le 
     //      // agrega un valor nuevo después o antes de la copia.
     //      // setCategories( [...categories, valor] ); 

     //      // otra forma con arrow function:
     //      setCategories( categories => [ ...categories, valor ] );
          
     // };

     return (
          <>
               <h1>GifExpert<span>App.</span></h1>
               <AddCategory setCategories={ setCategories }/>
               <ol>
                    {
                         categories.map( category => (
                              <GifGrid 
                                   key={ category }
                                   category={ category }
                              />
                         ) )
                    }
               </ol>
               <Footer />
          </>
     );
};

export default GifExpertApp;