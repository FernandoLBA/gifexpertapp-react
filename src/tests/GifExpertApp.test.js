import React from 'react';
import GifExpertApp from '../GifExpertApp';
import { shallow } from 'enzyme';

describe('Pruebas al componente <GifExpertApp />', () => {

     test('Debe hacer match el componente con el snapshot', () => {

          // Como sus props no son obligatorias podemos 
          // simplemente renderizarlo as√≠
          const wrapper = shallow( <GifExpertApp /> );

          expect( wrapper ).toMatchSnapshot();
     });

     test('Debe mostrar una lista de categorias', () => {
          const categories = [ 'Goku', 'Vegeta' ];

          const wrapper = shallow( <GifExpertApp defaultCategories={ categories } />);

          expect( wrapper ).toMatchSnapshot();

          expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );
     });
     
})
