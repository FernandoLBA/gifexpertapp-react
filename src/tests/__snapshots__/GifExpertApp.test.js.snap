// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifExpertApp /> Debe hacer match el componente con el snapshot 1`] = `
<Fragment>
  <h1>
    GifExpert
    <span>
      App.
    </span>
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
  <Footer />
</Fragment>
`;

exports[`Pruebas al componente <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpert
    <span>
      App.
    </span>
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
    <GifGrid
      category="Vegeta"
      key="Vegeta"
    />
  </ol>
  <Footer />
</Fragment>
`;
